flight/logger.py

```diff
@@ -0,0 +1,49 @@
+"""Logging configuration and functions"""
+import logging
+from typing import TextIO
+from datetime import datetime
+from multiprocessing import Queue
+from logging import Formatter, FileHandler, StreamHandler
+from logging.handlers import QueueHandler, QueueListener
+
+LOG_FILE: str = f"logs/{datetime.now()}.log"
+LOG_FORMAT: str = "%(levelname)s | %(asctime)s @  %(processName)s:%(funcName)s > %(message)s"
+LOG_LEVEL = logging.DEBUG
+
+
+def init_logger(queue: Queue[str]) -> QueueListener:
+    """
+    Creates a QueueListener that will process all log messages throughout the application
+    Parameters
+    ----------
+        queue: Queue
+            Queue object that holds logging processes
+    Returns
+    -------
+        QueueListener
+            Object to process log messages
+    """
+    console_formatter: Formatter = logging.Formatter(LOG_FORMAT)
+    file_formatter: Formatter = logging.Formatter(LOG_FORMAT)
+
+    file: FileHandler = logging.FileHandler(LOG_FILE, "a")
+    file.setFormatter(file_formatter)
+
+    console: StreamHandler[TextIO] = logging.StreamHandler()
+    console.setFormatter(console_formatter)
+
+    return QueueListener(queue, file, console)
+
+
+def worker_configurer(queue: Queue[str]) -> None:
+    """
+    When this is run, it configures the logger of this process to submit logs to the logging process (QueueListener)
```

Comment:
"When this is run"

I kind of assumed it only ran when it was run. (unnecessary phrase)